tls/x509utils/pem_write.go

```diff
@@ -0,0 +1,52 @@
+package x509utils
+
+import (
+	"bytes"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"io"
+
+	"darvaza.org/core"
+)
+
+// WriteKey writes a PEM encoded private key
+func WriteKey(w io.Writer, key PrivateKey) (int64, error) {
+	var buf bytes.Buffer
+
+	keyDER, err := x509.MarshalPKCS8PrivateKey(key)
```

Comment:
this `err` is shadowed so marshal errors will get wrapped as `failed to encode key`. This or today I am really bad at read code

Reply:
yes, same error wrapper for DER and PEM encoding errors. but the original is still available via `Unwrap()`.
I'm thinking `WriteKey()` and `WriteCert` only having two waits to fail. encoding and writing. what problem do you see on that rationale? @karasz